owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/ShareFileComposeFragment.kt

```diff
@@ -0,0 +1,626 @@
+                val capabilitiesUiResult = capabilitiesState.value?.peekContent()
+                val capabilities = capabilitiesUiResult?.getStoredData()
+                val isShareApiEnabled = capabilities?.filesSharingApiEnabled == CapabilityBooleanType.TRUE
+                val isPublicShareEnabled = capabilities?.filesSharingPublicEnabled == CapabilityBooleanType.TRUE
+                val isMultiplePublicSharingEnabled = capabilities?.filesSharingPublicMultiple?.isTrue ?: false
+
+                // Get shares to update some UI elements depending on them
+                val sharesState = ocShareViewModel.shares.observeAsState()
+                val sharesUiResult = sharesState.value?.peekContent()
+                val shares = sharesUiResult?.getStoredData()
+                var privateShares = shares?.filter { share ->
+                    share.shareType == ShareType.USER ||
+                            share.shareType == ShareType.GROUP ||
+                            share.shareType == ShareType.FEDERATED
+                }
+                var publicShares = shares?.filter { share ->
+                    share.shareType == ShareType.PUBLIC_LINK
+                }
+
+                when (capabilitiesUiResult) {
+                    is UIResult.Success -> {
+                        listener?.dismissLoading()
+                    }
+                    is UIResult.Error -> {
+                        capabilitiesState.value?.getContentIfNotHandled()?.let {
+                            showErrorInSnackbar(R.string.get_capabilities_error, capabilitiesUiResult.error)
+                        }
+                        listener?.dismissLoading()
+                    }
+                    is UIResult.Loading -> {
+                        listener?.showLoading()
+                    }
+                    else -> {
+                        // To avoid non-exhaustive when warning
+                    }
+                }
+
+                when (sharesUiResult) {
+                    is UIResult.Success -> {
+                        listener?.dismissLoading()
+                    }
+                    is UIResult.Error -> {
+                        sharesState.value?.getContentIfNotHandled()?.let {
+                            showErrorInSnackbar(R.string.get_shares_error, sharesUiResult.error)
+                        }
+                        listener?.dismissLoading()
+                    }
+                    is UIResult.Loading -> {
+                        listener?.showLoading()
+                    }
+                    else -> {
+                        // To avoid non-exhaustive when warning
+                    }
+                }
+
+                LazyColumn {
+                    item { SharedFileRow() }
+
+                    // Hide share with users section if it is not enabled or if share API is not enabled
+                    if (shareWithUsersAllowed && isShareApiEnabled) {
```

Comment:
Same as the previous `SharedFileRow` composable. We could extract the public shares section and the private shares section into self-composables. That way, we could maintain previews, or we could test the composable by itself.

For example:
- Preview or/and test for empty private shares
- Preview or/and test for not empty private shares
- Preview or/and test for empty public shares
- Preview or/and test for not empty public shares
- Preview or/and test for warning public shares

But since this one is the first iteration, should be fine 👍

Reply:
I've tried to modularise it as you suggest, but it's not as easy. Due to the use of ``items()`` in each of  the sections, I have to pass as parameter the LazyListScope of the LazyColumn, but it doesn't work well. This will be something to research, but for the moment I think it is ok